File: ladder-web/frontend/assets/js/modules/ai.js
// Модуль AI интеграции
import { api } from './api.js';

export async function decomposeTask(description) {
    try {
        const result = await api.decomposeTask(description);
        return result.subtasks || [];
    } catch (error) {
        console.error('Error decomposing task:', error);
        throw error;
    }
}

export function formatAISuggestions(subtasks) {
    return subtasks.map((subtask, index) => ({
        id: index + 1,
        title: subtask.title || subtask,
        description: subtask.description || '',
        priority: subtask.priority || 'medium',
    }));
}

File: ladder-web/backend/app/schemas/ai.py
"""
Pydantic схемы для AI
"""
from pydantic import BaseModel
from typing import List, Optional


class Subtask(BaseModel):
    title: str
    description: Optional[str] = None
    priority: Optional[str] = "medium"


class AIDecomposeRequest(BaseModel):
    description: str


class AIDecomposeResponse(BaseModel):
    subtasks: List[Subtask]
    original_description: str

